app.js

```diff
@@ -0,0 +1,490 @@
+        case 'generate_dataset':
+            func = router.generate_dataset;
+        case 'get_price_pred':
+            func = router.price_pred;
+    }
+  
+    try {
+        let res = await func.bind(router)(event.queryStringParameters);
+        return res;
+    } catch (err) {
+        console.log(err);
+        return err;
+    }
+}
+
+class Router
+{
+
+constructor()
+{
+this.alert_api = new WhaleAlertApi();
+this.db_handle = new DBStore();
+this.binance = new BinanceApi();
+this.ds = new Dataset( this.binance );
+this.predictor = new PricePredictor( this.binance, this.ds );
+}
+
+    respond( {body, type = "html", status = 200} )
+    {
+        if( type == "json" )
+        {
+            return { statusCode: status, headers: { 'Content-Type': 'application/json' }, body: body };
+        }
+        if( type == "html" )
+        {
+            return { statusCode: status, headers: { 'Content-Type': 'text/html; charset=utf-8' }, body: body };
+        }
+    }
+
+    async price_pred()
+    {
+        let res = await this.predictor.pred();
+        return this.respond( {body:JSON.stringify( res ), type:"text" } );
+    }
+
+    async generate_dataset()
+    {
+        let res = await this.binance.dataset_generator();
+        return this.respond( {body:JSON.stringify( res ), type:"text" } );
+    }
+
+    async get_page( params )
+    {
+        let source_path = config.STATIC_CONTENT_PATH[RUNTIME_ENV] + 'static/main.handlebars';
+        let source = await fs.readFile( source_path, "utf8");
+        return this.respond( {body:source } );;
+    }
+
+    // second end-point to fetch bucket data
+    async get_whale_buckets( params )
+    {
+        console.log( params );
+        let api_data = await this.db_handle.get_from_db( params );
+        let buckets =  this.alert_api.bucket_data_exchange( api_data );
+        if( !buckets ) return { statusCode: 500, body: 'Something wrong getting data buckets!' };
+        return this.respond( {body:JSON.stringify( buckets ), type:"json" } );
+    }
+
+    async update_db( params )
+    {
+        console.log( params );
+        let api_data = await this.alert_api.get_data( params );
+        
+        if( !api_data ) return { statusCode: 500, body: 'Error fetching data from downstream API!' };
+
+        let save_res = await this.db_handle.save_to_db( api_data );
+        
+        let count = api_data.transactions.length;
+        var response = this.respond( {body:'{"msg":"all '+count+' records saved to db"}', type:"json" } );
+        return response;
+    }
+}
+
+class Dataset
+{
+    constructor( web_api ) { this.web_api = web_api; }
+
+    async dataset_generator()
+     {
+        // 1 day => 1440 frames
+        let days = 1; // about 1sec per day? //37~49sec for 50days // 90sec for 100days
+        let winsize = 0; // 27 * (winsize + 1) 30 => 837 features
+        let whitelist = ["open", "close", "macd1_hist", "macd5_hist"];
+        let end_dateiso = "2021-01-17";
```

Comment:
- maybe can be const
- I would be more clear about the naming
- If this is business logic, now we need the comment on how the function works